### PR TITLE
refactor(web): folders store

### PR DESCRIPTION
**Pull request by michelheusschen**
Refactors the folders store to use runes

### CODE REVIEW

File: web/src/lib/stores/folders.svelte.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+   */
+  type AssetResponseDto,
+} from '@immich/sdk';
+
+type AssetCache = {
+  [path: string]: AssetResponseDto[];
+};
+
+class FoldersStore {
+  #initialized = false;
```
Comment (danieldietzler) [inline web/src/lib/stores/folders.svelte.ts:15]: Can this be private?

**Review by danieldietzler — APPROVED** [review]
Code looks good to me!

File: web/src/lib/stores/folders.svelte.ts (hunk 1 above)
Comment (michelheusschen) [reply web/src/lib/stores/folders.svelte.ts:15]: Sure

**Review by alextran1502 — APPROVED** [review]

File: web/tsconfig.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
     "skipLibCheck": true,
     "sourceMap": true,
     "strict": true,
-    "target": "es2020",
+    "target": "es2022",
```
[the remark this answers is not in the document]
Comment (akostadinov) [reply web/tsconfig.json:13]: I wonder why didn't upgrade to the latest :)

**Pull request merged**